### PR TITLE
Task on arrays

### PR DESCRIPTION
**Pull request by IgorSusolkin**

### CODE REVIEW

File: task on arrays/script.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+function isArray(obj) {
+  return typeof obj ==="object" &&
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: imo `typeof obj ==="object"` is not required
in additional code styles is bad. for your case you could use
```
return (typeof obj ==="object" &&
  Object.prototype.toString.call(obj) === "[object Array]");
```

File: task on arrays/script.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  Object.prototype.toString.call(obj) === "[object Array]";
+}
+function isArray2(obj) {
+  return !!obj && obj.constructor === Array;
+}
+function range(a, /*необязательный*/ b ,
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: `b,`

File: task on arrays/script.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  /*необязательный*/ c) {
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: it's not place for comments, doc string should be used

File: task on arrays/script.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+}
+function last(array) {
+  return array[array.length - 1];
+}
+function excludeLast(array, delLength) {
+  delLength = delLength || 1;
+  if (delLength > 0) {
+    if (delLength < array.length )
+    array.length -= delLength;
+    else array.length = 0;
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: initial array is mutated, -1
please use brackets for if else, it's better to read

File: task on arrays/script.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  return typeof obj ==="object" &&
+  Object.prototype.toString.call(obj) === "[object Array]";
+}
+function isArray2(obj) {
+  return !!obj && obj.constructor === Array;
+}
+function range(a, /*необязательный*/ b ,
+  /*необязательный*/ c) {
+  let first, last,
+    step = c || 1;
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: you could call c argument as step and don't define new variable

File: task on arrays/script.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    first = 0;
+    last = a;
+  }
+  let array = [];
+  for (let value = first; (step > 0) ? value < last :  value > last; value += step) {
+    array.push(value);
+  }
+  return array;
+}
+function compact(array) {
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: nit: please put empty line between functions, for readability

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: task on arrays/script.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  return toString.call(obj) === "[object Array]";
+}
+
+function isArray2(obj) {
+  return (!!obj && obj.constructor === Array);
+}
+
+function range(a, b, step) {
+  step = step || 1;
+  if (!b) {
```
Comment (NikitaSedyx) [inline task on arrays/script.js]: it won't work when b is 0
it's better to use strong condition in such cases

File: task on arrays/script.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
+function isArray(obj) {
+  return toString.call(obj) === "[object Array]";
+}
+
+function isArray2(obj) {
+  return (!!obj && obj.constructor === Array);
```
Comment (NikitaSedyx) [inline task on arrays/script.js:6]: nit: brackets are not required,  `obj` will work without type converting

**Review by NikitaSedyx — APPROVED** [review]
looks like it works, please fix comments

File: task on arrays/script.js (hunk 8 above)
Comment (IgorSusolkin) [reply task on arrays/script.js:6]: obj dont work without type converting
console.log(isArray2({}));//false
console.log(isArray2([]));//true
console.log(isArray2(""));//""
console.log(isArray2("5"));//false
console.log(isArray2(5));//false
console.log(isArray2(null));//null
console.log(isArray2(undefined));//underfined
console.log(typeof(isArray2({})));//boolean
console.log(typeof(isArray2([])));//boolean
console.log(typeof(isArray2("")));//string
console.log(typeof(isArray2("5")));//boolean
console.log(typeof(isArray2(5)));//boolean
console.log(typeof(isArray2(null)));//object
console.log(typeof(isArray2(undefined)));//undefined

**Pull request merged**